src/MongoDB.EntityFrameworkCore/Storage/MongoDatabaseWrapper.cs

```diff
@@ -327,6 +290,31 @@ private async Task<long> SaveMongoUpdatesAsync(IEnumerable<MongoUpdate> updates,
         return documentsAffected;
     }
 
+    private static BsonDocument ToBsonDocument(IUpdateEntry entry, IBsonDocumentSerializer entitySerializer, IEnumerable<IPropertyBase> properties)
```

Comment:
I propose refactoring `ToBsonDocument` to a slightly different method called `SerializeProperties`:

    private static void SerializeProperties(IBsonWriter writer, IUpdateEntry entry, IBsonDocumentSerializer entitySerializer, IEnumerable<IPropertyBase> properties)
    {
        var context = BsonSerializationContext.CreateRoot(writer);

        foreach (var property in properties)
        {
            var propertyValue = entry.GetCurrentValue(property);
            var propertySerializationInfo = GetPropertySerializationInfo(entitySerializer, property);
            var elementName = propertySerializationInfo.ElementName;
            var propertySerializer = propertySerializationInfo.Serializer;

            writer.WriteName(elementName);
            propertySerializer.Serialize(context, propertyValue);
        }
    }

Reply:
The name `SerializeProperties` emphasizes what this method *does* instead of simply naming the return type.

Also, instead of returning a `BsonDocument` it takes a `writer`. This allows us to build up a document by passing the same `writer` to several methods.